apps/api/src/common/hotcache.ts

```diff
@@ -0,0 +1,34 @@
+class CacheObject {
+  constructor(public value: any, public expiration: number) {}
+  isExpired() {
+    return this.expiration < Date.now()
+  }
+}
+
+export class HotCache {
+  constructor(private ttl: number, private enabled: boolean) {}
+  private cache = new Map<string, any>()
+
+  lookup(key: string) {
+    if (!this.enabled) {
+      return null
+    }
+
+    const cacheObject = this.cache.get(key)
+    if (cacheObject && !cacheObject.isExpired()) {
+      return cacheObject.value
+    }
+    return null
+  }
+
+  // make the ttl optional
+  store(key: string, value: any, ttl?: number) {
```

Comment:
value: any raises a linter error due to noExplicitAny. 
In this context the safer type is 'unknown' as we don't know the type of the stored value, however at the place where used the type is known so type-checking will work well